### DISCUSSION TITLE
Permissions issues

### DISCUSSION THREAD

**Discussion by vivinano** [Rapptz/discord.py#9178] [Q&A]
I keep trying to set roles using bot but it saying I don't have permissions even tho I sent intents to true and within the developer portal set the options to true 

When I do a print discord.permissions it is still set at 0

**Comment by cibere** [comment]
Intents and permissions are 2 separate things. Intents are global, and it tells discord what data to send to your bot. Permissions is per guild, and limits what your bot can and can't do.

If `discord.Permissions` is set to 0, then your bot has no permissions. As mentioned in the [docs on this](https://discordpy.readthedocs.io/en/latest/api.html?highlight=add_roles#discord.Member.add_roles),
> You must have [manage_roles](https://discordpy.readthedocs.io/en/latest/api.html?highlight=add_roles#discord.Permissions.manage_roles) to use this, and the added [Role](https://discordpy.readthedocs.io/en/latest/api.html?highlight=add_roles#discord.Role)s must appear lower in the list of roles than the highest role of the member.

**Reply by vivinano** [reply]
Okay so first I need to set intents
Is there a way to confirm this 

Second I need to check permissions in the client side ? And then the development portal side ?

**Comment by vivinano** [comment]
![image](https://user-images.githubusercontent.com/33008397/212501977-8523dd67-89b4-4b24-b473-51c07be8afb7.png)


![image](https://user-images.githubusercontent.com/33008397/212501999-751f853e-1a11-4792-8d01-669c7e95c284.png)

Discord.pemissions()
Discord.pemissions().manage_roles
Intents 
![image](https://user-images.githubusercontent.com/33008397/212502006-1d660f34-b024-42bd-af60-7793376be5e9.png)

**Reply by cibere** [reply]
Once again, intents and permissions are 2 completely separate items. Regarding the [permissions.manage_roles](https://discordpy.readthedocs.io/en/latest/api.html?highlight=permissions#discord.Permissions.manage_roles) returning false however, you are creating a brand new permissions object via [discord.Permissions()](https://discordpy.readthedocs.io/en/latest/api.html?highlight=permissions#discord.Permissions). And by default, it has no permissions, which would explain it having `value` set to `0`. To get an instance of a permissions object which has your bots permissions, you first have to get the guild, then you can use [guild.me](https://discordpy.readthedocs.io/en/latest/api.html?highlight=guild%20me#discord.Guild.me) to get your bots member object, then [member.guild_permissions](https://discordpy.readthedocs.io/en/latest/api.html?highlight=guild%20me#discord.Member.guild_permissions) returns a permissions object that contains your bots permissions.

If this is confusing, consider learning OOP (object oriented programming) before attempting to use discord.py
